tools/update_runnablejobs.py

```diff
@@ -243,6 +226,49 @@ def add_jobs_to_jobpriority(new_data=None, priority=1, timeout=0, set_expired=Fa
                           .values(buildsystem=_buildsystem)
             conn.execute(statement)
 
+
+def add_jobs_to_jobpriority(new_data=None, priority=1, timeout=0, set_expired=False):
+    if not new_data:
+        return
+
+    # This is a loop to remove duplicates (including buildsystem -> * if needed)
+    # By doing this it allows us to have a single database query
+    # no_duplicates will contain a subset which excludes:
+    #   * jobs that don't specify the build_platform
+    #   * jobs that don't specify the testtype
+    #   * XXX: unclear why we append the job again but with *
+    no_duplicates = []
+    for job in new_data['results']:
+        platform = parse_platform(job['build_platform'])
+        testtype = parse_testtype(job['build_system_type'],
+                                  job['ref_data_name'],
+                                  job['platform_option'],
+                                  job['job_type_name'])
+        if not platform or not testtype:
+            continue
+
+        job["platform"] = platform
+        job["testtype"] = testtype
+        # Let's see if we need to append the job with '*' as the build_system_type
+        for nd in no_duplicates:
+            if job['testtype'] == nd['testtype'] and
+               job['platform'] == nd['platform'] and
+               job['platform_option'] == nd['platform_option']:
+
+                if job['build_system_type'] != nd['build_system_type']:
+                    # This will make so we add the same job but with '*'
+                    # XXX: What is the purpose of this?
+                    job['build_system_type'] = '*'
+                break
+
+        no_duplicates.append(job)
+
+    # XXX: Does this affect the new_data outside of this function?
+    new_data['results'] = no_duplicates
+
+    added_jobs = _update_database_with_new_jobs(new_data=new_data)
+
+    # XXX: We don't actually use this information outside of the function
```

Comment:
correct- possibly we don't need to return it- I had returned the information for debugging purposes while developing it.